code/game/objects/structures/lavaland/ore_vent.dm

```diff
@@ -0,0 +1,502 @@
+	return ..()
+
+/obj/structure/ore_vent/Destroy()
+	SSore_generation.possible_vents -= src
+	node = null
+	if(tapped)
+		SSore_generation.processed_vents -= src
+	return ..()
+
+/obj/structure/ore_vent/attackby(obj/item/attacking_item, mob/user, params)
+	. = ..()
+	if(.)
+		return
+	if(!is_type_in_list(attacking_item, scanning_equipment))
+		return
+	if(tapped)
+		balloon_alert_to_viewers("vent tapped!")
+		return TRUE
+	scan_and_confirm(user)
+	return TRUE
+
+/obj/structure/ore_vent/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	if(.)
+		return
+	if(!isgolem(user))
+		return
+	if(!discovered)
+		to_chat(user, span_notice("You can't quite find the weakpoint of [src]... Perhaps it needs to be scanned first?"))
+		return TRUE
+	to_chat(user, span_notice("You start striking [src] with your golem's fist, attempting to dredge up a boulder..."))
+	for(var/i in 1 to 3)
+		if(do_after(user, boulder_size * 1 SECONDS, src))
+			user.apply_damage(20, STAMINA)
+			playsound(src, 'sound/weapons/genhit.ogg', 50, TRUE)
+	produce_boulder()
+	visible_message(span_notice("You've successfully produced a boulder! Boy are your arms tired."))
+	return TRUE
+
+/obj/structure/ore_vent/is_buckle_possible(mob/living/target, force, check_loc)
+	. = ..()
+	if(tapped)
+		return FALSE
+	if(istype(target, /mob/living/basic/node_drone))
+		return TRUE
+
+/obj/structure/ore_vent/examine(mob/user)
+	. = ..()
+	if(discovered)
+		switch(boulder_size)
+			if(BOULDER_SIZE_SMALL)
+				. += span_notice("This vent produces [span_bold("small")] boulders containing [ore_string]")
+			if(BOULDER_SIZE_MEDIUM)
+				. += span_notice("This vent produces [span_bold("medium")] boulders containing [ore_string]")
+			if(BOULDER_SIZE_LARGE)
+				. += span_notice("This vent produces [span_bold("large")] boulders containing [ore_string]")
+	else
+		. += span_notice("This vent can be scanned with a [span_bold("Mining Scanner")].")
+
+/obj/structure/ore_vent/add_context(atom/source, list/context, obj/item/held_item, mob/living/user)
+	if(is_type_in_list(held_item, scanning_equipment))
+		context[SCREENTIP_CONTEXT_LMB] = "Scan vent"
+		return CONTEXTUAL_SCREENTIP_SET
+
+/**
+ * Picks n types materials to pack into a boulder created by this ore vent, where n is this vent's minerals_per_boulder.
+ * Then assigns custom_materials based on boulder_size, assigned via the ore_quantity_function
+ */
+/obj/structure/ore_vent/proc/create_mineral_contents()
+
+	var/list/refined_list = list()
+	for(var/iteration in 1 to minerals_per_boulder)
+		var/datum/material/material = pick_weight(mineral_breakdown)
+		refined_list.Insert(refined_list.len, material)
+		refined_list[material] += ore_quantity_function(iteration)
+	return refined_list
+
+/obj/structure/ore_vent/proc/generate_mineral_breakdown(max_minerals = MINERAL_TYPE_OPTIONS_RANDOM, map_loading = FALSE)
+	var/iterator = 1
+	if(max_minerals < 1)
+		CRASH("generate_mineral_breakdown called with max_minerals < 1.")
+	while(iterator <= max_minerals)
```

Comment:
You could just use a for loop.
`for(var/iterator = 1; iterator <= max_minerals; iterator++)`